typed-racket-doc/typed-racket/scribblings/guide/caveats.scrbl

```diff
@@ -180,3 +180,66 @@ circumvent the protections of the type system.
 
 Contract boundaries installed for typed-untyped interaction may cause
 significant slowdowns. See @secref{contract-costs} for details.
+
+@section{Pattern Matching and Occurrence Typing}
+
+
+Because Typed Racket type checks code @emph{after} macro
+expansion, certain forms---such as @racket[match]---are
+difficult for Typed Racket to completely reasoning about. In
```

Comment:
"reasoning" => "reason"

Reply:
If you're picky about split infinitives (I'm not) you could say "to reason about completely" or "to understand fully" or whatever.

Reply:
Thanks